### PR TITLE
fix make:sonata:admin command

### PR DESCRIPTION
**Pull request by OskarStark**
I am targeting this branch, because this is BC.

Rework of #5252 (thank you @volodymyr-nakvasiuk)

## Changelog

```markdown
### Fixed
- `make:sonata:admin` not working
```

## Subject

I changed `RuntimeException` to `InvalidArgumentException` which could be considered as a BC break, but imo its better to use it here.

I get this error when using `if (!\count($this->availableManagers))`:
screenshot 2018-10-10 19 34 12

//cc @indiancomet

### CODE REVIEW

File: src/Maker/AdminMaker.php [hunk 1]
```diff
@@ -282,8 +282,8 @@ private function configure(InputInterface $input)
             $this->controllerClassBasename = Validators::validateControllerClassBasename($this->controllerClassBasename);
         }
 
-        if (!\count($this->availableModelManagers)) {
-            throw new \RuntimeException('There are no model managers registered.');
+        if (0 == \count($this->availableModelManagers)) {
```
Comment (core23) [inline src/Maker/AdminMaker.php]: ===?

Comment (OskarStark) [reply src/Maker/AdminMaker.php]: sure thing

Comment (OskarStark) [reply src/Maker/AdminMaker.php]: done

**Comment by axzx** [comment]
This is not solution to the problem. I have the same error:
```
In AdminMaker.php line 286:
                                           
  There are no model managers registered.
```

Test:
```
dump(0 === \count($this->availableModelManagers));
dump(!\count($this->availableModelManagers));
```

return:
```
true
true
```

**Comment by axzx** [comment]
I think the problem is here:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/SonataAdminBundle.php#L48

test:
```
dump($container->has('MakerBundle'));
```

result:
```
false
```

**Comment by axzx** [comment]
besides, the `AdminMaker` class has a lot of errors

**Comment by axzx** [comment]
Yes, #5252 fix my problem.

```
symfony/*                                          v4.1.6
symfony/maker-bundle                     v1.7.0
```

**Comment by OskarStark** [comment]
Great would you like to rework #5252 and fix the tests?

**Comment by axzx** [comment]
@OskarStark  I do not feel strong yet to do that

**Comment by OskarStark** [comment]
I added tests

**Comment by volodymyr-nakvasiuk** [comment]
Thanks, @OskarStark for finishing!

File: src/Maker/AdminMaker.php [hunk 2]
```diff
@@ -282,8 +283,8 @@ private function configure(InputInterface $input)
             $this->controllerClassBasename = Validators::validateControllerClassBasename($this->controllerClassBasename);
         }
 
-        if (!\count($this->availableModelManagers)) {
-            throw new \RuntimeException('There are no model managers registered.');
+        if (0 === \count($this->availableModelManagers)) {
+            throw new \InvalidArgumentException('There are no model managers registered.');
```
Comment (core23) [inline src/Maker/AdminMaker.php]: Why did you change the exception type?

Comment (OskarStark) [reply src/Maker/AdminMaker.php]: See PR header

**Comment by tmaslov22** [comment]
I still have this problem on sonata-project/admin-bundle 3.68 and symfony 4.3

**Comment by MagePsycho** [comment]
Still, there's the same issue with Symfony 5 (with master branch)
